lib/ruby_lsp/requests/completion.rb

```diff
@@ -39,6 +39,7 @@ def initialize(index, nesting, emitter, message_queue)
         @_response = T.let([], ResponseType)
         @index = index
         @nesting = nesting
+        @typechecker = T.let(DependencyDetector.instance.typechecker?, T::Boolean)
```

Comment:
I'm wondering whether it's worth storing this in an instance var, or just calling the singleton each time.